airbyte-protocol/models/src/main/resources/airbyte_protocol/airbyte_message.yaml

```diff
@@ -118,10 +118,43 @@ definitions:
       name:
         type: string
         description: Stream's name.
-      json_schema:
-        description: Stream schema using Json Schema specs.
+      discovered:
+        description: Contains information about the data in the source. This field is set by the source and should be read-only thereafter.
         type: object
-        existingJavaType: com.fasterxml.jackson.databind.JsonNode
+        properties:
+          json_schema:
+            description: Stream schema using Json Schema specs.
+            type: object
+            existingJavaType: com.fasterxml.jackson.databind.JsonNode
+          metadata:
+            description: Additional information about the data in the source, including information about what Airbyte features it supports.
+            type: object
+            properties:
+              supported_sync_modes:
+                type: array
+                items:
+                  "$ref": "#/definitions/SyncMode"
+                # backwards compatibility: if not present, assume full_refresh only.
+                default:
+                  [full_refresh]
+              default_comparable:
+                description: Path to the field that will be used to determine if a record is new or modified since the last sync. If not provided by the source, the end user will have to specify the comparable themselves.
+                type: array
+                items: string
+      configuration:
+        description: Contains information about how the data should be synced. this section is configured by the end-user and is never touched by the source.
+        properties:
+          # backwards compatibility: if not present, assume full_refresh.
+          sync_mode:
+            "$ref": "#/definitions/SyncMode"
+            default: full_refresh
+          comparable:
+            description: Path to the field that will be used to determine if a record is new or modified since the last sync. This field is REQUIRED if `sync_mode` is `incremental`. Otherwise it is ignored.
```

Comment:
This is the path from the catalog, not the message, right? Might be worth noting that.

Reply:
not sure i understand. let's do an example.
* source: postgres
* table name: users
* column to be used for comparable: updated_at.

I imagine the comparable field would be `["updated_at"]` _not_ `["users", "updated_at"]`. which one did you think it was? But yes, I can clarify this.

Reply:
I agree, `["updated_at"]` makes sense. I just want that to be obvious after reading the description (or maybe the desc can include this example).

Reply:
still need to document this when i bring the docs back. later PR at this point. made a note.